pkg/maps/tunnel/tunnel.go

```diff
@@ -50,23 +55,72 @@ func init() {
 
 // +k8s:deepcopy-gen=true
 // +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
-// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapValue
 type TunnelEndpoint struct {
 	bpf.EndpointKey
 }
 
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapValue
+type TunnelEndpointInfo struct {
```

Comment:
This seems to duplicate:
https://github.com/cilium/cilium/blob/master/pkg/bpf/endpoint.go#L24-L30

Is there a way to avoid this duplication? Hopefully one that would allow us to also not duplicate code for some of the functions such as `NewEndpointKey`, `ToIP`, and `String()`.

For example, maybe it would make sense to have `EndpointKey` embedded in `TunnelEndpointInfo`?

Reply:
the  lxcmap  also use the NewEndpointKey
https://github.com/cilium/cilium/blob/master/pkg/maps/lxcmap/lxcmap.go#L143-L157
the intention of this PR is to introduce minimal code changes to not have regression, I agree there are some code duplications, even in lxcmap.go, I am still not fluent in cilium golang code enough to do the refactoring, and most important, to introduce regression :) let me know what you think.

Reply:
@kkourt  actually, endpointKey **ToIP**, **String** are used in lxcmap https://github.com/cilium/cilium/blob/master/pkg/maps/lxcmap/lxcmap.go#L242, it looks to me each map key and value structure requires their own **ToIP**, **String** because they may have different fields, so from that point of new, **ToIP**, **String**  are not duplicates. what do you think?

Reply:
Hopefully there is not too much boilerplate from this, just a few extra lines; for stuff like `String()` I'd expect that we have different implementations since the key / value should actually have different fields, like the ipsec key is actually part of the value and is never used for lookup.